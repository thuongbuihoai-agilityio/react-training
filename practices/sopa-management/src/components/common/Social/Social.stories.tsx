import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';

// Components
import Social from '@common/Social';

const meta: Meta<typeof Social> = {
  component: Social,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <Story />
      </BrowserRouter>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Social>;

export const SocialList: Story = {
  args: {}
};
