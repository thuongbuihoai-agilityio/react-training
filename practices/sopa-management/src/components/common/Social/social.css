.social-item {
  display: flex;
  margin-bottom: var(--m-medium);
}

.social-link {
  margin-right: var(--m-regular);
}
