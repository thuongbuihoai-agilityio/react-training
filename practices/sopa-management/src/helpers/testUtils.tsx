import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import {
  QueryClient,
  QueryClientProvider
} from 'react-query';
import {
  BrowserRouter,
  Router
} from 'react-router-dom';

export const renderRouterTest = (component: JSX.Element) => {
  const history = createMemoryHistory();
  return render(
    <Router location={history.location} navigator={history}>
      {component}
    </Router>
  );
};

export const renderWithRouterAndQuery = (children: JSX.Element) => {
  const queryClient = new QueryClient();

  return render(
    <BrowserRouter>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </BrowserRouter>
  );
};
