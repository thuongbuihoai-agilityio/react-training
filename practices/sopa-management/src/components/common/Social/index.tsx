import { Link } from 'react-router-dom';

// Components
import {
  Icon,
  IconType,
  Text,
  ThemeType
} from '@components/common';

// Styles
import './social.css';

const Social = () => (
  <ul data-testid='social' className='social-list'>
    <li className='social-item'>
      <Link
        title='twitter'
        to='https://twitter.com'
        className='social-link'
        aria-label='link to twitter'
      >
        <Icon iconName={IconType.twitter} />
      </Link>
      <Text text='Twitter' type={ThemeType.light} />
    </li>
    <li className='social-item'>
      <Link
        title='instagram'
        to='https://instagram.com'
        className='social-link'
        aria-label='link to instagram'
      >
        <Icon iconName={IconType.instagram} />
      </Link>
      <Text text='Instagram' type={ThemeType.light} />
    </li>
    <li className='social-item'>
      <Link
        title='facebook'
        to='https://facebook.com'
        className='social-link'
        aria-label='link to facebook'
      >
        <Icon iconName={IconType.facebook} />
      </Link>
      <Text text='Facebook' type={ThemeType.light} />
    </li>
  </ul>
);

export default Social;
