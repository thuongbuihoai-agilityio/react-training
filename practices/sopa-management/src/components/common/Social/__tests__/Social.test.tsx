// Libs
import '@testing-library/jest-dom';

// Helpers
import { renderRouterTest } from '@helpers/testUtils';

// Components
import Social from '@common/Social';

describe('Social component', () => {
  test('should render Social component', () => {
    const { getByTestId } = renderRouterTest(<Social />);
    expect(getByTestId('social')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = renderRouterTest(<Social />);
    expect(asFragment()).toMatchSnapshot();
  });
});
