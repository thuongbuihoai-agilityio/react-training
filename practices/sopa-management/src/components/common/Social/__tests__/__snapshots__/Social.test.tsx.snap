// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social component matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="social-list"
    data-testid="social"
  >
    <li
      class="social-item"
    >
      <a
        aria-label="link to twitter"
        class="social-link"
        href="https://twitter.com"
        title="twitter"
      >
        <div
          class="undefined icon-twitter"
          data-testid="icon"
        />
      </a>
      <p
        class=" text-light"
        data-testid="text"
      >
        Twitter
      </p>
    </li>
    <li
      class="social-item"
    >
      <a
        aria-label="link to instagram"
        class="social-link"
        href="https://instagram.com"
        title="instagram"
      >
        <div
          class="undefined icon-instagram"
          data-testid="icon"
        />
      </a>
      <p
        class=" text-light"
        data-testid="text"
      >
        Instagram
      </p>
    </li>
    <li
      class="social-item"
    >
      <a
        aria-label="link to facebook"
        class="social-link"
        href="https://facebook.com"
        title="facebook"
      >
        <div
          class="undefined icon-facebook"
          data-testid="icon"
        />
      </a>
      <p
        class=" text-light"
        data-testid="text"
      >
        Facebook
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
